netmap/netmap_contract.go

```diff
@@ -281,13 +281,17 @@ func UpdateState(state int, publicKey interop.PublicKey) {
 	if notaryDisabled {
 		alphabet = common.AlphabetNodes()
 		nodeKey = common.InnerRingInvoker(alphabet)
-		if len(nodeKey) == 0 {
-			common.CheckWitness(publicKey)
+	}
 
-			runtime.Notify("UpdateState", state, publicKey)
-			return
-		}
+	// If notary is enabled or caller is not an alphabet node,
+	// just emit the notification for alphabet.
+	if !notaryDisabled || len(nodeKey) == 0 {
+		common.CheckWitness(publicKey)
+		runtime.Notify("UpdateState", state, publicKey)
```

Comment:
why have you decided to produce a notification in notary-enabled mode?

Reply:
To mirror `AddPeer` behaviour.

Reply:
Agree that those notifications in notary enabled environment are kinda useless. Let's remove them and put useful notifications directly to `removeFromNetmap` and `addToNetmap` functions.

Reply:
Fixed, removed.